From 65437d233438b9c9b14309162a67f5df10dd0e47 Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Mon, 2 Nov 2015 11:41:02 +0800
Subject: [PATCH 1206/1221] MLK-11803 pci: imx: imx6qp pcie ep self-test
 failed

imx pcie used the wrab mode to do the cached access
methods on axi bus. There is 64bytes address mis-aligned
problem.
Disable the cached operations.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
(cherry picked from commit 85db70336ab66136481926bcd7f5abe599e2aa4f)
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/pci/host/pci-imx6.c |    6 +-----
 1 files changed, 1 insertions(+), 5 deletions(-)

diff --git a/drivers/pci/host/pci-imx6.c b/drivers/pci/host/pci-imx6.c
index 2ddbbf0..df8ab44 100644
--- a/drivers/pci/host/pci-imx6.c
+++ b/drivers/pci/host/pci-imx6.c
@@ -1413,11 +1413,7 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 		 * reserved ddr memory after write the ddr_test_region
 		 * content to rc.
 		 */
-		if (is_imx7d_pcie(imx6_pcie))
-			pcie_arb_base_addr = ioremap_nocache(pp->mem_base,
-					test_region_size);
-		else
-			pcie_arb_base_addr = ioremap_cache(pp->mem_base,
+		pcie_arb_base_addr = ioremap_nocache(pp->mem_base,
 					test_region_size);
 
 		if (!pcie_arb_base_addr) {
-- 
1.7.5.4

